wallet/src/main/java/com/android/identity_credential/wallet/presentation/OnlinePresentmentActivity.kt

```diff
@@ -0,0 +1,821 @@
+import com.nimbusds.jwt.EncryptedJWT
+import com.nimbusds.jwt.JWTClaimsSet
+import com.nimbusds.jwt.SignedJWT
+import com.nimbusds.jwt.proc.BadJWTException
+import com.nimbusds.jwt.proc.DefaultJWTProcessor
+import com.nimbusds.jwt.proc.JWTClaimsSetVerifier
+import io.ktor.client.HttpClient
+import io.ktor.client.call.NoTransformationFoundException
+import io.ktor.client.call.body
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.client.request.accept
+import io.ktor.client.request.get
+import io.ktor.client.request.post
+import io.ktor.http.ContentType
+import io.ktor.http.HttpStatusCode
+import io.ktor.http.Parameters
+import io.ktor.http.Url
+import io.ktor.http.content.OutgoingContent
+import io.ktor.http.formUrlEncode
+import io.ktor.serialization.kotlinx.json.*
+import io.ktor.util.InternalAPI
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.suspendCancellableCoroutine
+import kotlinx.coroutines.withContext
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+import kotlinx.serialization.Required
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonElement
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.json.JsonPrimitive
+import kotlinx.serialization.json.buildJsonObject
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.encodeToJsonElement
+import kotlinx.serialization.json.jsonArray
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import kotlinx.serialization.json.put
+import java.security.cert.X509Certificate
+import java.util.UUID
+import kotlin.coroutines.resume
+import kotlin.io.encoding.Base64
+import kotlin.io.encoding.ExperimentalEncodingApi
+import kotlin.random.Random
+
+@Serializable
+data class PresentationSubmission(
+    @Required val id: String,
+    @Required
+    @SerialName("definition_id")
+    val definitionId: String,
+    @Required
+    @SerialName("descriptor_map")
+    val descriptorMaps: List<DescriptorMap>,
+) : java.io.Serializable
+
+@Serializable
+data class DescriptorMap(
+    @Required val id: String,
+    @Required val format: String,
+    @Required val path: String,
+) : java.io.Serializable
+
+class AuthorizationRequest (
+    var presentationDefinition: JsonObject,
+    var clientId: String,
+    var nonce: String,
+    var responseUri: String,
+    var state: String?,
+    var clientMetadata: JsonObject,
+    var certificateChain: List<X509Certificate>?
+)
+
+class ResponseComponents (
+    var documentRequest: DocumentRequest,
+    var sessionTranscript: ByteArray,
+    var jweHeader: JWEHeader,
+    var clientId: String,
+    var responseUrl: Url,
+    var state: String?,
+    var presentationSubmission: PresentationSubmission,
+    val keySet: JWKSet
+)
+
```

Comment:
I think these classes can all be `private` or `internal`. Any reason the latter two aren't data classes?